iio/iio.c

```diff
@@ -1141,7 +1141,22 @@ static uint32_t iio_generate_device_xml(struct iio_device *device, char *name,
 		buff = ch_id;
 
 	i = 0;
-	i += snprintf(buff, max(n - i, 0),
+
+	/* Write context attributes */
+	if (device->context_attributes)
+		for (j = 0; device->context_attributes[j].name; j++) {
+			i += snprintf(buff + i,
+				      max(n - i, 0),
+				      "<context-attribute name=\"%s\" ",
+				      device->context_attributes[j].name);
```

Comment:
I'm wondering if it'd make sense to put some default values for .name and .value but I guess it's also fine without any defaults

Reply:
@buha - sorry, is this question for me or other reviewers? Where do you suggest default values, I didn't get you?

Reply:
I was just thinking out loud. I think currently the name defaults to "no-OS" and the value is the git version that defaults to something like "1.0.0-g0000000". If device->context_attributes[j].name is NULL, I think this is undefined behaviour in C:
https://stackoverflow.com/questions/11589342/what-is-the-behavior-of-printing-null-with-printfs-s-specifier

So maybe just initialize device->context_attributes[j].name with "no-OS" if it is NULL, prior to the snprintf call.

Reply:
Looks like NULL check done in the for loop. Would that be good?
```
for (j = 0; device->context_attributes[j].name; j++)
```

Reply:
seems ok then, just make sure to fix conflicts